src/include/hardware.h

```diff
@@ -49,21 +48,24 @@
         }                                                                                          \
     } while (0)
 #else
-#define WT_RELEASE_WRITE(v, val) __atomic_store_n(&(v), (val), __ATOMIC_RELEASE);
-#endif
-#else
-#define WT_RELEASE_WRITE(v, val) WT_RELEASE_WRITE_WITH_BARRIER(v, val);
+#define WT_RELEASE_WRITE(v, val) __atomic_store_n(&(v), (val), __ATOMIC_RELEASE)
 #endif
 
 /*
  * Deprecated: use WT_RELEASE_WRITE instead. Release write a value to a shared location. All
  * previous stores must complete before the value is made public.
+ *
+ * Use atomic based operation for TSAN since it doesn't track standalone barriers synchronization
```

Comment:
Nitpick: some grammar fixups.
* Missing fullstop
* `operation` -> `operations`
* Maybe drop the word `synchronization`?

Same for below, although that one has a copy-paste error with an extra asterisk.